### PR TITLE
Allow ascii whitespace char for doc aliases 

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes issue from https://github.com/rust-lang/rust/issues/76705#issuecomment-703123847

cc @lopopolo @ollie27 

r? @jyn514

### CODE REVIEW

**Comment by lopopolo** [comment]
Thank you! 📚

**Comment by jyn514** [comment]
You need to run `x.py test src/test/rustdoc --test-args check-doc-alias --bless`:
```
---- [ui] ui/check-doc-alias-attr.rs stdout ----
diff of stderr:

36	LL | | ")]
37	   | |_^
38	
-	error: ' ' character isn't allowed in `#[doc(alias = "...")]`
-	  --> $DIR/check-doc-alias-attr.rs:14:7
-	   |
-	LL | #[doc(alias = " ")]
-	   |       ^^^^^^^^^^^
-	
45	error: '\t' character isn't allowed in `#[doc(alias = "...")]`
46	  --> $DIR/check-doc-alias-attr.rs:15:7
47	   |

48	LL | #[doc(alias = "\t")]
49	   |       ^^^^^^^^^^^^
50	
-	error: aborting due to 8 previous errors
+	error: aborting due to 7 previous errors
```
Other than that this seems fine to me.

**Comment by jyn514** [comment]
Actually ollie had concerns about this in https://github.com/rust-lang/rust/issues/76705#issuecomment-693020017, so

r? @ollie27

File: compiler/rustc_passes/src/check_attr.rs [hunk 1]
```diff
@@ -287,8 +287,9 @@ impl CheckAttrVisitor<'tcx> {
                             self.doc_alias_str_error(meta);
                             return false;
                         }
-                        if let Some(c) =
-                            doc_alias.chars().find(|&c| c == '"' || c == '\'' || c.is_whitespace())
+                        if let Some(c) = doc_alias
+                            .chars()
+                            .find(|&c| c == '"' || c == '\'' || (c.is_whitespace() && c != ' '))
```
Comment (pickfire) [inline compiler/rustc_passes/src/check_attr.rs:292]: Why not change it to `is_ascii_whitespace`?

Comment (GuillaumeGomez) [reply compiler/rustc_passes/src/check_attr.rs:292]: Because I assume it includes `\t`, `\r` and `\n`, which we still don't want.

Comment (pickfire) [reply compiler/rustc_passes/src/check_attr.rs:292]: Yes, it is there.
```rust
        match *self {
            '\t' | '\n' | '\x0C' | '\r' | ' ' => true,
            _ => false,
        }
```

**Comment by ollie27** [comment]
The issue I had with whitespace is that search queries are trimmed so if an alias begins with or ends in whitespace then it's impossible to search for it. We could allow spaces as long as they're not at the beginning or end of the alias though.

**Comment by GuillaumeGomez** [comment]
Excellent point! Then I'll update this PR to take this into account.

**Comment by GuillaumeGomez** [comment]
Updated!

File: compiler/rustc_passes/src/check_attr.rs [hunk 2]
```diff
@@ -303,6 +303,16 @@ impl CheckAttrVisitor<'tcx> {
                                 .emit();
                             return false;
                         }
+                        if doc_alias.starts_with(' ') || doc_alias.ends_with(' ') {
```
Comment (pickfire) [inline compiler/rustc_passes/src/check_attr.rs:306]: What if people put `\t`?

Comment (jyn514) [reply compiler/rustc_passes/src/check_attr.rs:306]: `\t` is handled above on line 292, it's not allowed at all.

**Comment by GuillaumeGomez** [comment]
ping @ollie27

**Comment by GuillaumeGomez** [comment]
I fixed the last comments so let's move on!

@bors: r=jyn514,ollie27

**Pull request merged**